### ISSUE TITLE
apex mixed precision with horovod

### ISSUE THREAD

**Issue by gongjingcs** [horovod/horovod#2282]
hi，
I noticed that horovod must be used like this style：
     # Apex
    with amp.scale_loss(loss, optimizer) as scaled_loss:
        scaled_loss.backward()
        optimizer.synchronize()
    with optimizer.skip_synchronize():
        optimizer.step()
in combination with apex in your example code https://gist.github.com/alsrgv/0713add50fe49a409316832a31612dde#file-pytorch_synthetic_benchmark_apex-py-L85 ? Can you explain in detail why？ thanks alot
